addons/FluxibleMixin.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2014, Yahoo! Inc.
+ * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+ */
+'use strict';
+
+/**
+ * React mixin for staticly declaring and add/remove-ing listeners for Store events.
+ * @class FluxibleMixin
+ * @example
+ * // Register listener default handler function name
+ * var Component = React.createClass({
+ *     mixins: [FluxibleMixin],
+ *     statics: {
+ *         storeListeners: [FooStore]
+ *     },
+ *     onChange: function () {
+ *         this.setState(this.context.getStore(FooStore).getState());
+ *     },
+ *     ...
+ * });
+ * @example
+ * // Register listener with custom named handler
+ * var Component = React.createClass({
+ *     mixins: [FluxibleMixin],
+ *     statics: {
+ *         storeListeners: {
+ *             'onChange2': [FooStore]
+ *         }
+ *     },
+ *     onChange2: function () {
+ *         this.setState(this.context.getStore(FooStore).getState());
+ *     },
+ *     ...
+ * });
+ */
+var DEFAULT_CHANGE_HANDLER = 'onChange';
+var React = require('react');
+
+var FluxibleMixin = {
+
+    contextTypes: {
+        getStore: React.PropTypes.func,
+        executeAction: React.PropTypes.func
+    },
+
+    /**
+     * Registers staticly declared listeners
+     * @method componentDidMount
+     */
+    componentDidMount: function componentDidMount() {
+        this.listeners = [];
+        var self = this;
+
+        // Register static listeners
+        this.getListeners().forEach(function(listener) {
+            self._attachStoreListener(listener);
+        });
+    },
+
```

Comment:
can we provide a default `onChange` method on this mixin?  It could be something like:

```js
onChange: function() {
   // The user could need to either define the getStateFromStore method or override onChange
   this.setState(this.getStateFromStore());
}
```